### ISSUE TITLE
in_kafka_group swallows error in run and stops reading input

### ISSUE THREAD

**Issue by sgran** [fluent/fluent-plugin-kafka#77]
```
def run
  @consumer.each_batch(@fetch_opts) { |batch|
  ...
  rescue => e
    $log.error "unexpected error", :error => e.to_s
    $log.error_backtrace
  end
end
```

This logic breaks out of the consumer.each_batch loop, and does not resume it.  We had a kafka cluster failover this morning, resulting in the following stacktrace:

```
  2016-10-03 09:07:16 +0000 [error]: unexpected error error="Kafka::NotLeaderForPartition"
  2016-10-03 09:07:16 +0000 [error]: /opt/td-agent/embedded/lib/ruby/gems/2.1.0/gems/ruby-kafka-0.3.12/lib/kafka/protocol.rb:50:in `handle_error'
  2016-10-03 09:07:16 +0000 [error]: /opt/td-agent/embedded/lib/ruby/gems/2.1.0/gems/ruby-kafka-0.3.12/lib/kafka/fetch_operation.rb:73:in `block (3 levels) in execute'
  2016-10-03 09:07:16 +0000 [error]: /opt/td-agent/embedded/lib/ruby/gems/2.1.0/gems/ruby-kafka-0.3.12/lib/kafka/fetch_operation.rb:71:in `map'
  2016-10-03 09:07:16 +0000 [error]: /opt/td-agent/embedded/lib/ruby/gems/2.1.0/gems/ruby-kafka-0.3.12/lib/kafka/fetch_operation.rb:71:in `block (2 levels) in execute'
  2016-10-03 09:07:16 +0000 [error]: /opt/td-agent/embedded/lib/ruby/gems/2.1.0/gems/ruby-kafka-0.3.12/lib/kafka/fetch_operation.rb:70:in `each'
  2016-10-03 09:07:16 +0000 [error]: /opt/td-agent/embedded/lib/ruby/gems/2.1.0/gems/ruby-kafka-0.3.12/lib/kafka/fetch_operation.rb:70:in `flat_map'
  2016-10-03 09:07:16 +0000 [error]: /opt/td-agent/embedded/lib/ruby/gems/2.1.0/gems/ruby-kafka-0.3.12/lib/kafka/fetch_operation.rb:70:in `block in execute'
  2016-10-03 09:07:16 +0000 [error]: /opt/td-agent/embedded/lib/ruby/gems/2.1.0/gems/ruby-kafka-0.3.12/lib/kafka/fetch_operation.rb:59:in `each'
  2016-10-03 09:07:16 +0000 [error]: /opt/td-agent/embedded/lib/ruby/gems/2.1.0/gems/ruby-kafka-0.3.12/lib/kafka/fetch_operation.rb:59:in `flat_map'
  2016-10-03 09:07:16 +0000 [error]: /opt/td-agent/embedded/lib/ruby/gems/2.1.0/gems/ruby-kafka-0.3.12/lib/kafka/fetch_operation.rb:59:in `execute'
  2016-10-03 09:07:16 +0000 [error]: /opt/td-agent/embedded/lib/ruby/gems/2.1.0/gems/ruby-kafka-0.3.12/lib/kafka/consumer.rb:244:in `fetch_batches'
  2016-10-03 09:07:16 +0000 [error]: /opt/td-agent/embedded/lib/ruby/gems/2.1.0/gems/ruby-kafka-0.3.12/lib/kafka/consumer.rb:160:in `block in each_batch'
  2016-10-03 09:07:16 +0000 [error]: /opt/td-agent/embedded/lib/ruby/gems/2.1.0/gems/ruby-kafka-0.3.12/lib/kafka/consumer.rb:195:in `consumer_loop'
  2016-10-03 09:07:16 +0000 [error]: /opt/td-agent/embedded/lib/ruby/gems/2.1.0/gems/ruby-kafka-0.3.12/lib/kafka/consumer.rb:159:in `each_batch'
  2016-10-03 09:07:16 +0000 [error]: /opt/td-agent/embedded/lib/ruby/gems/2.1.0/gems/fluent-plugin-kafka-0.3.1/lib/fluent/plugin/in_kafka_group.rb:120:in `run'
```

No more messages were consumed after this.

I think the rescue block probably wants to retry, or perhaps reraise the exception so that fluentd can retry the input as a whole.

Thanks,

**Comment by repeatedly** [comment]
Thanks for the report.
You are right. in_kafka_group should re-run consumer.
I will fix it.

**Comment by repeatedly** [comment]
Released v0.3.2
